ingestion/src/metadata/ingestion/models/custom_pydantic.py

```diff
@@ -37,6 +45,29 @@ class BaseModel(PydanticBaseModel):
     Specified as `--base-class BASE_CLASS` in the generator.
     """
 
+    @model_validator(mode="after")
+    @classmethod
+    def parse_name(cls, values):  # pylint: disable=inconsistent-return-statements
+        """
+        Primary entry point to process values based on their class.
+        """
+
+        if not values:
+            return
+
+        try:
+
+            if cls.__name__ in CREATE_ADJACENT_MODELS or cls.__name__.startswith(
+                "Create"
+            ):
+                values = validate_name_and_transform(values, replace_separators)
+            elif cls.__name__ in FETCH_MODELS:
+                values = validate_name_and_transform(values, revert_separators)
+
+        except Exception as exc:
+            logger.warning("Exception while parsing special characters: %s", exc)
```

Comment:
prefer to re-raise and crash. otherwise the system is in an unknown state.

Reply:
re raised and crashed, if it faces any error